### PR TITLE
Fix #1331 : In scene, add OR condition to check-time action

### LINKED ISSUE #1331: Scene condition: Schedule from 10pm to 4am not working
When I want to add a time condition like "execute between 10pm and 4am every day" it did not work on certain conditions.

* From 10pm to 11:55pm ✅
* From 12am to 5am ✅
* From 10pm to 5am ❌

### PR DESCRIPTION
**Pull request by Pierre-Gilles**
### Pull Request check-list

To ensure your Pull Request can be accepted as fast as possible, make sure to review and check all of these items:

- [x] If your changes affects code, did your write the tests?
- [x] Are tests passing? (`npm test` on both front/server)
- [x] Is the linter passing? (`npm run eslint` on both front/server)
- [x] Did you run prettier? (`npm run prettier` on both front/server)
- [x] If you are adding a new features/services, did you run integration comparator? (`npm run compare-translations` on front)
- [x] Did you test this pull request in real life? With real devices? If this development is a big feature or a new service, we recommend that you provide a Docker image to [the community](https://community.gladysassistant.com/) for testing before merging.
- [x] If your changes modify the API (REST or Node.js), did you modify the API documentation? (Documentation is based on comments in code)
- [x] If you are adding a new features/services which needs explanation, did you modify the user documentation? See [the GitHub repo](https://github.com/GladysAssistant/v4-website) and the [website](https://gladysassistant.com).
- [x] Did you add fake requests data for the demo mode (`front/src/config/demo.js`) so that the demo website is working without a backend? (if needed) See [https://demo.gladysassistant.com](https://demo.gladysassistant.com).

NOTE: these things are not required to open a PR and can be done afterwards / while the PR is open.

### Description of change

This let the user use conditional time check when after time is before the before time.

### CODE REVIEW

**Review by cicoub13 — APPROVED** [review]
Code is OK (thanks for the Boy Scout rule application on tests 😉 )

Tested locally with some functional cases => OK for me

```
2022-01-31T13:54:37+0100 <debug> scene.actions.js:161 (Object.condition.check-time) Check time before: 13:54 < 09:10 condition is not verified.
2022-01-31T13:54:37+0100 <debug> scene.actions.js:176 (Object.condition.check-time) Check time after: 13:54 > 13:35 condition is valid.
2022-01-31T13:54:37+0100 <debug> scene.actions.js:192 (Object.condition.check-time) Check time: Condition OR verified.
```

**Comment by Pierre-Gilles** [comment]
Thanks for the review @cicoub13 ! 🙏 What's the "Boy Scout rule application on tests" ? 😂

**Comment by cicoub13** [comment]
Boy Scout Rule in IT = "Always leave the code better than you found it"

You used this fix opportunity to move test code in dedicated file 💪 

https://biratkirat.medium.com/step-8-the-boy-scout-rule-robert-c-martin-uncle-bob-9ac839778385

**Comment by Pierre-Gilles** [comment]
Oh yes indeed 👍

File: server/test/lib/scene/actions/scene.action.checkTime.test.js [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+              type: ACTIONS.CONDITION.CHECK_TIME,
+              after: sevenPm.format('HH:mm'),
+              before: nineAm.format('HH:mm'),
+            },
+          ],
+        ],
+        scope,
+      );
+    } catch (e) {
+      console.log(e);
```
Comment (atrovato) [inline server/test/lib/scene/actions/scene.action.checkTime.test.js:153]: Unexpected console statement!

**Review by atrovato — APPROVED** [review]
Code is good enough to me, I didn't test it in real life, I suppose others will ;)

**Pull request merged**